src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
Builder throw IndexOutOfRangeException and ArgumentOutOfRangeExceptionexception exception. Now it throw only ArgumentOutOfRangeExceptionexception exception. I fixed next issue `The indexer doesn't check for index being larger than or equal to 0, resulting in accessing the underlying array with a negative index, causing IndexOutOfRangeException. Note that List<T> does a cast to uint as part of the bounds check to cover this case.`